openlibrary/plugins/openlibrary/dev_instance.py

```diff
@@ -160,3 +177,79 @@ def load_sample_data():
     comment = "Loaded sample data."
     list_key = "/people/anand/lists/OL1815L"
     env['copy_list'](src, dest, list_key, comment=comment)
+
+
+def start_debugger():
+    """ Attach debugger and immediately break into debugger """
+
+    logger.debug("------------> ATTEMPTING TO ENABLE DEBUGGING <--------------")
+    try:
+        sys.path.append("/usr/local/lib/python2.7/dist-packages")
+        ptvsd = import_module("ptvsd")
+        ptvsd.enable_attach(address=('0.0.0.0', 3000))
+        ptvsd.wait_for_attach()
+        ptvsd.break_into_debugger()
+        sys.path.pop()
+    except Exception as e:
+        logger.error("something bad happened {}".format(str(e)))
+    logger.debug("-----------------> DEBUGGING ENABLED <----------------------")
+
+
+def permission(key, readers, writers, admins):
+    """ Query for permission type \"thing\"s """
+
+    def make_key(value):
+        return {'key': value}
+
+    return {
+        'key': key,
+        'type': {
+            'key': '/type/permission'
+        },
+        'readers': [make_key(readers)],
+        'writers': [make_key(writers)],
+        'admins': [make_key(admins)]
+    }
+
+
+def root_page(key):
+    """ Query for root page type \"thing\" """
+    return {
+        'key': key,
+        'type': {
+            'key': '/type/page'
+        },
+        'permission': {
+            'key': '/permission/restricted'
+        },
+        'child_permission': {
+            'key': '/permission/loggedinusers'
+        }
+    }
+
+
+@infogami.install_hook
+def add_missing_metadata():
+    """ Add missing /permission types and root pages """
+
+    logger.debug("Adding missing permissions")
+    for missing_permission in ["/permission", "/permission/loggedinusers"]:
```

Comment:
It would be better to use a tuple instead of a list because 1) takes a bit less RAM and 2) signals that the values should not be modified at runtime.

Reply:
good catch